libs/vkd3d/swapchain.c

```diff
@@ -1905,13 +1968,19 @@ static void dxgi_vk_swap_chain_present_callback(void *chain_)
     if (!chain->wait_thread.active)
         present_count = 1;
 
+    if (chain->queue->device->vk_info.NV_low_latency2)
+        pthread_mutex_lock(&chain->present.low_latency_lock);
```

Comment:
Taking a lock here is very bad since present_iteration() can block.
It seems more useful to latch any state quickly and then avoid locks later.

Reply:
The low latency lock will only be taken when the swapchain is destroyed now.